src/core/match_data.rs

```diff
@@ -0,0 +1,100 @@
+use super::LangData;
+use std::{
+    iter::Sum,
+    ops::{Add, AddAssign},
+    path::PathBuf,
+};
+
+/// Data structure representing a match found by one or more heuristics.
+#[derive(Debug, Clone)]
+pub struct MatchData {
+    /// Path of the matched file.
+    pub path: PathBuf,
+    /// Path of the directory which was being processed while the match was found.
+    /// 
+    /// Used to group multiple results in a meaningful way.
+    pub group: PathBuf,
+    /// Additional data about the match, queried with implemented methods.
+    pub(super) other_data: MatchDataBuilder,
```

Comment:
wait? `MatchData` have in itself `MatchDataBuilder`? Isn't the `MatchDataBuilder` suppose to create instances of `MatchData`?

Reply:
This is made in order to avoid many slightly different structs that only differ by their way of storing paths. I have decided that this way `MatchData` just contains all the fields that it needs by storing the builder itself, and they are all accessible by dedicated getters (no setters because you shouldn't need to change fields of this struct).

Reply:
Ok, so as i can see those two structs can be merged into one - on every step of existnce `MatchDataBuilder` you know `path` and `group`. So all methed's form `MatchDataBuilder` can be moved to `MatchData` without intoducing any unnesesery `Option<>` types. That will siplify structure of `MatchData` (don't require geters) incresing readeblity.
The lame option is to rename it to somethink that isn't call builder - because it's not a [builder](https://en.wikipedia.org/wiki/Builder_pattern)!